misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java

```diff
@@ -86,4 +89,24 @@ public static boolean isBlank(String str) {
         }
         return srcNames;
     }
+
+    /**
+     * Get relative path when a bal file is inside a directory.
+     *
+     * @param sourcePath     source path
+     * @param patternMatcher pattern matcher string
+     * @param sourceName     source name
+     * @param fileName       file name
+     * @return relative path
+     */
+    public static String getDirectoryRelativePath(String sourcePath, String patternMatcher, String sourceName,
+                                                  String fileName) {
+        Pattern pattern = Pattern.compile(SERVICE_REGEX);
+        Matcher matcher = pattern.matcher(patternMatcher);
+        if (matcher.find()) {
+            return sourcePath;
+        } else {
+            return sourcePath.replace(sourceName, fileName);
+        }
+    }
```

Comment:
@praveennadarajah Shall we add a comment explaining that in here we have handled the service files specifically and the reason behind that?

Reply:
Added comments.

Reply:
Looks good!